heimlig/src/hsm/keystore.rs

```diff
@@ -106,50 +134,117 @@ impl<const STORAGE_SIZE: usize, const NUM_KEYS: usize> KeyStore
     for MemoryKeyStore<STORAGE_SIZE, NUM_KEYS>
 {
     fn is_stored(&self, id: KeyId) -> bool {
-        match self.infos.iter().find(|key_info| key_info.id == id) {
+        match self
+            .layout
+            .iter()
+            .find(|key_layout| key_layout.info.id == id)
```

Comment:
Does it make sense to store only the sorted layout and use the binary search to improve the performance? https://doc.rust-lang.org/nightly/core/primitive.slice.html#method.binary_search